From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: The-going <48602507+The-going@users.noreply.github.com>
Date: Sat, 8 Feb 2025 12:37:21 +0300
Subject: arm64: sun50i-h616: Add i2c(2,3,4), uart(2,5) pins

Add a description of the pins for further use
in device trees and overlays.

link to:
drivers/pinctrl/sunxi/pinctrl-sun50i-h616.c
---
 arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi | 103 ++++++++++
 1 file changed, 103 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
@@ -366,11 +366,48 @@ i2c0_pins: i2c0-pins {
 				function = "i2c0";
 			};
 
+			/omit-if-no-ref/
+			i2c2_ph_pins: i2c2-ph-pins {
+				pins = "PH2", "PH3";
+				function = "i2c2";
+			};
+
+			/omit-if-no-ref/
+			i2c2_pi_pins: i2c2-pi-pins {
+				pins = "PI9", "PI10";
+				function = "i2c2";
+			};
+
+			i2c3_pa_pins: i2c3-pa-pins {
+				pins = "PA10", "PA11";
+				function = "i2c3";
+				bias-pull-up;
+			};
+
+			/omit-if-no-ref/
+			i2c3_pg_pins: i2c3-pg-pins {
+				pins = "PG17", "PG18";
+				function = "i2c3";
+			};
+
+			/omit-if-no-ref/
 			i2c3_ph_pins: i2c3-ph-pins {
 				pins = "PH4", "PH5";
 				function = "i2c3";
 			};
 
+			/omit-if-no-ref/
+			i2c4_pg_pins: i2c4-pg-pins {
+				pins = "PG15", "PG16";
+				function = "i2c4";
+			};
+
+			/omit-if-no-ref/
+			i2c4_ph_pins: i2c4-ph-pins {
+				pins = "PH6", "PH7";
+				function = "i2c4";
+			};
+
 			ir_rx_pin: ir-rx-pin {
 				pins = "PH10";
 				function = "ir_rx";
@@ -484,6 +521,72 @@ uart1_rts_cts_pins: uart1-rts-cts-pins {
 				function = "uart1";
 			};
 
+			/omit-if-no-ref/
+			uart2_pg_pins: uart2-pg-pins {
+				pins = "PG15", "PG16";
+				function = "uart2";
+			};
+
+			/omit-if-no-ref/
+			uart2_pg_rts_cts_pins: uart2-pg-rts-cts-pins {
+				pins = "PG17", "PG18";
+				function = "uart2";
+			};
+
+			/omit-if-no-ref/
+			uart2_ph_pins: uart2-ph-pins {
+				pins = "PH5", "PH6";
+				function = "uart2";
+			};
+
+			/omit-if-no-ref/
+			uart2_ph_rts_cts_pins: uart2-ph-rts-cts-pins {
+				pins = "PH7", "PH8";
+				function = "uart2";
+			};
+
+			/omit-if-no-ref/
+			uart2_pi_pins: uart2-pi-pins {
+				pins = "PI5", "PI6";
+				function = "uart2";
+			};
+
+			/omit-if-no-ref/
+			uart2_pi_rts_cts_pins: uart2-pi-rts-cts-pins {
+				pins = "PI7", "PI8";
+				function = "uart2";
+			};
+
+			/omit-if-no-ref/
+			uart3_pi_pins: uart3-pi-pins {
+				pins = "PI9", "PI10";
+				function = "uart2";
+			};
+
+			/omit-if-no-ref/
+			uart3_pi_rts_cts_pins: uart3-pi-rts-cts-pins {
+				pins = "PI11", "PI12";
+				function = "uart2";
+			};
+
+			/omit-if-no-ref/
+			uart4_pi_pins: uart4-pi-pins {
+				pins = "PI13", "PI14";
+				function = "uart2";
+			};
+
+			/omit-if-no-ref/
+			uart4_pi_rts_cts_pins: uart4-pi-rts-cts-pins {
+				pins = "PI15", "PI16";
+				function = "uart2";
+			};
+
+			/omit-if-no-ref/
+			uart5_pins: uart5-pins {
+				pins = "PH2", "PH3";
+				function = "uart5";
+			};
+
 			/omit-if-no-ref/
 			x32clk_fanout_pin: x32clk-fanout-pin {
 				pins = "PG10";
-- 
Armbian

